bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/handler/SiemensHvacBridgeBaseThingHandler.java

```diff
@@ -0,0 +1,187 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SiemensHvacBridgeBaseThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Laurent Arnal - Initial contribution and API
+ */
+@NonNullByDefault
+public abstract class SiemensHvacBridgeBaseThingHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SiemensHvacBridgeBaseThingHandler.class);
+    private @Nullable SiemensHvacDeviceDiscoveryService discoveryService;
+    private final @Nullable HttpClientFactory httpClientFactory;
+    private final SiemensHvacMetadataRegistry metaDataRegistry;
+    private @Nullable SiemensHvacBridgeConfig config;
+
+    public SiemensHvacBridgeBaseThingHandler(Bridge bridge, @Nullable NetworkAddressService networkAddressService,
+            @Nullable HttpClientFactory httpClientFactory, SiemensHvacMetadataRegistry metaDataRegistry) {
+        super(bridge);
+        SiemensHvacConnector lcConnector = null;
+        this.httpClientFactory = httpClientFactory;
+        this.metaDataRegistry = metaDataRegistry;
+
+        lcConnector = this.metaDataRegistry.getSiemensHvacConnector();
+        if (lcConnector != null) {
+            lcConnector.setSiemensHvacBridgeBaseThingHandler(this);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        metaDataRegistry.invalidate();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "@text/offline.waiting-bridge-initialization");
+
+        SiemensHvacBridgeConfig lcConfig = getConfigAs(SiemensHvacBridgeConfig.class);
+        String baseUrl = null;
+
+        baseUrl = lcConfig.baseUrl;
+
+        if (baseUrl.isEmpty()) {
+            logger.debug("@text/offline.error-gateway-init");
+            return;
+        }
+
+        if (!baseUrl.startsWith("http://") && !baseUrl.startsWith("https://")) {
+            baseUrl = "http://" + baseUrl;
+        }
+
+        if (!baseUrl.endsWith("/")) {
+            baseUrl = baseUrl + "/";
+        }
+
+        config = lcConfig;
+
+        // Will read metadata in background to not block initialize for a long period !
+        scheduler.schedule(this::initializeCode, 1, TimeUnit.SECONDS);
+    }
+
+    protected String getBuildDate() {
+        try {
+            ClassLoader cl = getClass().getClassLoader();
+            if (cl != null) {
+                URL res = cl.getResource(getClass().getCanonicalName().replace('.', '/') + ".class");
+                if (res != null) {
+                    URLConnection cnx = res.openConnection();
+                    Date dt = new Date(cnx.getLastModified());
```

Comment:
Can you use one of the newer `java.time` Java classes here, for example `LocalDate`?

Reply:
fix in next commit